batch/batch/server/server.py

```diff
@@ -1068,6 +1071,19 @@ def __anext__(self):
                 log.exception(f'Delete pvc {pvc.metadata.name} failed due to exception: {e}')
 
 
+async def create_pods_if_ready():
```

Comment:
I'm slightly confused why this is necessary.  Won't this be done by `refresh_k8s_pods`?  If you don't have a pod it will call `update_job_with_pod(job, None)` which in turn calls `mark_unscheduled` which creates the pod.

Reply:
My understanding of `refresh_k8s_pods` is it does the following:

- Gets jobs where the pod is not None
a. if that pod is in k8s, then update the job with the pod state
b. if pod is not in k8s, update job with pod = None

However, it doesn't handle the case where the state is Ready but pod=None such as if a pvc failed to create. So I thought I needed another polling loop to do this. I didn't see another place in the code where we'd attempt to recreate the pod in this scenario.